repository/src/fetch.rs

```diff
@@ -0,0 +1,269 @@
+use super::*;
+use thiserror::Error;
+
+async fn add_remotes<T: RawRepository>(
+    this: &mut DistributedRepository<T>,
+    known_peers: &[Peer],
+) -> Result<(), Error> {
+    for peer in known_peers {
+        let remote_name = peer.name.clone();
+        let remote_url = format!("git://{}/repo", peer.address.ip());
```

Comment:
is this git's configuration or ours?

Reply:
You mean `/repo`? That's our protocol. See the next PR.